client/extensions/woocommerce/state/sites/promotions/handlers.js

```diff
@@ -77,3 +91,46 @@ export function couponsUpdated( { dispatch }, action ) {
 	}
 }
 
+export function promotionCreate( { dispatch }, action ) {
+	const { siteId, promotion } = action;
+
+	if ( 'coupon' === promotion.type && promotion.coupon ) {
+		dispatch( createCoupon( siteId, promotion.coupon, action.successAction, action.failureAction ) );
+	}
+	if ( 'product_sale' === promotion.type && promotion.product ) {
+		dispatch( updateProduct( siteId, promotion.product, action.successAction, action.failureAction ) );
+	}
+}
+
+export function promotionUpdate( { dispatch }, action ) {
+	const { siteId, promotion } = action;
+
+	if ( 'coupon' === promotion.type && promotion.coupon ) {
+		dispatch( updateCoupon( siteId, promotion.coupon, action.successAction, action.failureAction ) );
+	}
+	if ( 'product_sale' === promotion.type && promotion.product ) {
+		dispatch( updateProduct( siteId, promotion.product, action.successAction, action.failureAction ) );
+	}
+}
+
+export function promotionDelete( { dispatch }, action ) {
+	const { siteId, promotion } = action;
+
+	if ( 'coupon' === promotion.type && promotion.coupon ) {
+		dispatch( deleteCoupon( siteId, promotion.coupon.id, action.successAction, action.failureAction ) );
+	}
+	if ( 'product_sale' === promotion.type && promotion.product ) {
+		// Remove all sale-related fields from the product.
+		const {
+			sale_price,
+			date_on_sale_from,
+			date_on_sale_from_gmt,
+			date_on_sale_to,
+			date_on_sale_to_gmt,
+			...productDeletedSale
+		} = promotion.product; // eslint-disable-line no-unused-vars
```

Comment:
Could we use https://lodash.com/docs/4.17.4#omit instead? I find it clearer to read/grok.